api-bucket-lifecycle.go

```diff
@@ -24,14 +24,15 @@ import (
 	"io"
 	"net/http"
 	"net/url"
+	"strconv"
 	"time"
 
 	"github.com/minio/minio-go/v7/pkg/lifecycle"
 	"github.com/minio/minio-go/v7/pkg/s3utils"
 )
 
 // SetBucketLifecycle set the lifecycle on an existing bucket.
-func (c *Client) SetBucketLifecycle(ctx context.Context, bucketName string, config *lifecycle.Configuration) error {
+func (c *Client) SetBucketLifecycle(ctx context.Context, bucketName string, config *lifecycle.Configuration, expiryRuleRemoved bool) error {
```

Comment:
No breaking APIs please @shtripat :-) 

Please add a new API or move `expiryRuleRemoved` into `lifecycle.Configuration` as another variable.

Reply:
I think I will move this under `lifecycle.Configuration` to keep it non breaking.